cli/Tests/TuistKitTests/Services/GenerateServiceTests.swift

```diff
@@ -159,11 +162,215 @@ struct GenerateServiceTests {
                 includedTargets: [],
                 noOpen: false,
                 configuration: nil,
-                ignoreBinaryCache: false
+                ignoreBinaryCache: false,
+                cacheProfile: nil
             )
 
             // Then
             try TuistTest.expectLogs("Total time taken: 0.234s")
         }
     }
+
+    @Test func passes_allPossible_when_targets_focused_overrides_explicit_profile() async throws {
+        // Given
+        let workspacePath = try AbsolutePath(validating: "/test.xcworkspace")
+        let environment = MapperEnvironment()
+        given(configLoader).loadConfig(path: .any).willReturn(
+            .test(project: .testGeneratedProject())
+        )
+        given(generator)
+            .generateWithGraph(path: .any, options: .any)
+            .willReturn(
+                (
+                    workspacePath,
+                    .test(),
+                    environment
+                )
+            )
+
+        // When
+        try await subject.run(
+            path: nil,
+            includedTargets: ["App"],
+            noOpen: true,
+            configuration: nil,
+            ignoreBinaryCache: false,
+            cacheProfile: "none"
+        )
+
+        // Then
+        verify(generatorFactory)
+            .generation(
+                config: .any,
+                includedTargets: .value(["App"]),
+                configuration: .any,
+                cacheProfile: .matching { $0 == .init(base: .allPossible, targets: []) },
+                cacheStorage: .any
+            )
+            .called(1)
+    }
+
+    @Test func passes_explicit_builtin_profile_all_possible() async throws {
```

Comment:
This is not in line with the current behavior. `tuist generate` should default to only external when no focused target is passed. The motivation behind it is that in development, developers likely want just the external dependencies to be turned into binaries.